lib/internal/test_runner/test.js

```diff
@@ -649,6 +650,10 @@ class Test extends AsyncResource {
         this.reporter.coverage(this.nesting, kFilename, this.coverage);
       }
 
+      // In case the event loop has ended and reporter has not drained,
+      // we use a timer to keep the process alive until the reporter is done.
+      const handle = setInterval(() => {}, TIMEOUT_MAX);
+      this.reporter.once('close', () => clearInterval(handle));
```

Comment:
How is the stream still working but not keeping the process alive? Is there a resource somewhere that we `unref()`d ?

Reply:
see https://github.com/nodejs/node/issues/22088#issuecomment-1436097725
the reporters transform the output after the child process are done

Reply:
I am bisecting now

Reply:
Interesting, investigating then, could you guide me if at any place in the test_runner code where pipeline maybe being used?

Reply:
I assume the timing condition we might be facing is within `setupTestReporters`
where `getReportersMap` is fulfilled after all the tests completed running
https://github.com/nodejs/node/blob/5e954c3ce3cce314ef8b96abd5948298c0603d4d/lib/internal/test_runner/utils.js#L151-L158

Reply:
Ah understood, thank you checking this

Reply:
For the record: I think the issue is in the test runner, not in the streams

Reply:
So far no success in thinking with reference to https://github.com/nodejs/node/commit/b4a962d0e6728157983c709398321fd03e902db0 only think that comes to mind is if anyway testsStream ended before somehow? but that also doesn't hold up much 😕 nonetheless will try explore a little more just incase